friends/friendController.js

```diff
@@ -0,0 +1,86 @@
+const router = require('express').Router();
+const Friend = require('./friendModel.js');
+
+
+router
+	.route('/')
+	.get((req, res) => {
+		console.log(Friend.find);
+		Friend.find()
+			.then(friends => {
+				res.status(200).json(friends);
+  			})
+			.catch(err => res.status(500).json({ error: "The information could not be retrieved."	}))
+	})
+
+	.post((req, res) => {
+		const { firstName, lastName, age } = req.body
+		const newFriend = new Friend({ firstName, lastName, age });
+		if (!req.body.firstName || !req.body.lastName || !req.body.age) {
+			res.status(400).json({ error: "Please provide first name, last name, and age for your friend."});
+// all 3 fields are required
+		} else if (typeof age !=='number' || age < 1 || age > 120) {
+			res.status(400).json({ error: "Age must be a number between 1 and 120." });
+		}
+// make sure age is actually a number AND that it is between 1 and 120		
+		else {
+			newFriend.save()
+				.then(Friend => {
+					res.status(201).json({ Friend });
+				})
+				.catch(err => res.status(500).json({ error: "There was an error saving the friend to the database." }))
+		}
+	})
+router
+	.route('/:id')
+	.get((req, res) => {
+	const { id } = req.params;
+		Friend.findById(id)
+			.then(foundFriend => {
+				if (!foundFriend) {
+					res.status(404).json({ error: "The friend with the specified ID could not be retrieved." });
+				}
+				else {
+					res.status(200).json(foundFriend);
+				}
+			})
+	  		.catch(err => res.status(500).json({ error: "There was an error retrieving your friend." }));
+	})
+// I tried to do a better job of getting the 404 and 500 messages for this project - yesterday I spent so much time figuring out why my thing wasn't working (it was a missing "s") that I didn't get this quite right.
+
```

Comment:
Those little things will getcha every time! You error messages are good. I don't know if you caught this in the CS10 channel, but Dre'Sean shared a cool trick: if you use ({ error: error.message }) you get a more descriptive error message in your json response (or console.log statement).